### ISSUE TITLE
C domain generation doesn't work with Breathe v4.12+

### ISSUE THREAD

**Issue by rdeterre** [michaeljones/breathe#483]
Hi,

When using the `breathe_domain_by_extension` variable in conf.py to set the domain of some files to be C, the documentation generated for enums is invalid. This is only true for Breathe versions v4.12 and up, v4.11 works OK.

The gist of it is that the enum definition below generates the output from the image copied below, where the enumerand names are missing.

``` c
/// A dummy enum
enum color {
    red   = 0xFF0000,
    green = 0x00FF00,
    blue  = 0x0000FF,
};
```

![image](https://user-images.githubusercontent.com/3432890/76247679-500c3000-6216-11ea-8c51-75ee648bfeb4.png)

Here is a link to a minimal example that reproduces this issue: https://github.com/rdeterre/breatherepro

My setup for testing was the following:

- Windows machine
- Python 3.8.1
- Sphinx 2.4.1

I tried all releases of Breathe since v4.11.

**Comment by melvinvermeeren** [comment]
Possibly related to 4557198a42dedaa4255ee7398ce17c26391fc4a7 and/or df5b389cc66d68d355ac40d9e30cc15b84b6b97b.

If you have some time to help could you check the diffs and locally test with modifications in these parts to see what fixes it?

**Comment by vermeeren** [comment]
Might be automatically fixed with Sphinx 3 and latest Breathe, considering the C domain was rewritten and Breathe now uses Sphinx in a more proper way compared to before.

If you could verify that it would be great.

**Comment by vermeeren** [comment]
Should be fixed with #512, released in Breathe v4.17.0. Note that you also need Sphinx 3.x for recent Breathe versions.
